pkg/quota/v1/evaluator/core/resource_claims.go

```diff
@@ -187,6 +189,16 @@ func (p *claimEvaluator) getVerifiedPodUsage(claim *resourceapi.ResourceClaim) c
 	if pod.Status.ExtendedResourceClaimStatus != nil && pod.Status.ExtendedResourceClaimStatus.ResourceClaimName != claim.Name {
 		return nil
 	}
+	if pod.Status.ExtendedResourceClaimStatus == nil {
+		key := claim.Namespace + "/" + string(pod.UID)
+		if c, ok := p.podToExtendedResourceClaimCache[key]; ok && (c.Namespace != claim.Namespace || c.Name != claim.Name) {
+			return nil
+		}
+		p.podToExtendedResourceClaimCache[key] = &resourceapi.ResourceClaim{
```

Comment:
mutating this isn't thread-safe

we also need to populate this with other claims we observe, otherwise an N-server cluster can admit up to N resourceclaims before they notice conflicts

I think an index that keeps track of resource claims we've seen by owning pod uid would be more correct cross-server

https://github.com/liggitt/kubernetes/commits/resource-claim-conflict-quota/ was an (untested) POC in that direction